shotover/src/transforms/kafka/sink_cluster/scram_over_mtls.rs

```diff
@@ -108,21 +109,34 @@ async fn task(
         tokio::select! {
             biased;
             username = recreate_queue.next() => {
+                let old_token = username_to_token.get(&username).cloned();
                 let instant = Instant::now();
-                let token = create_token::create_token_with_timeout(
+                let new_token = create_token::create_token_with_timeout(
                     &mut nodes,
                     &mut rng,
                     mtls_connection_factory,
                     &username,
                     delegation_token_lifetime
                 ).await
                 .with_context(|| format!("Failed to recreate delegation token for {:?}", username))?;
-                username_to_token.insert(username.clone(), token);
+
+                username_to_token.insert(username.clone(), new_token);
```

Comment:
`.insert` will return the old token, so lets do `let old_token = username_to_token.insert(...`